src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2022,16 +2045,232 @@ private IOperation PeekOperand()
+                                        AddPlaceholder(argumentPlaceholder, capture);
+                                    }
+                                    else
+                                    {
+                                        // Can happen if we're in an indexer object initializer (an error scenario where the instance isn't yet available).
+                                        Debug.Assert(capturedInstance == null);
+                                        AddPlaceholder(argumentPlaceholder, new InvalidOperation(
+                                            ImmutableArray<IOperation>.Empty,
+                                            semanticModel: null,
+                                            syntax: argumentPlaceholder.Syntax,
+                                            type: argumentPlaceholder.Type,
+                                            constantValue: argumentPlaceholder.GetConstantValue(),
+                                            isImplicit: true));
+                                    }
+
+                                    placeholderKeys.Add(argumentPlaceholder);
+                                    break;
+
+                                case InterpolatedStringArgumentPlaceholderKind.CallsiteArgument:
+                                    if (argumentPlaceholder.ArgumentIndex >= i)
+                                    {
+                                        // Out of position argument: just capture an invalid operation and reference it from the constructor
+                                        AddPlaceholder(argumentPlaceholder, new InvalidOperation(
+                                            ImmutableArray<IOperation>.Empty,
+                                            semanticModel: null,
+                                            syntax: argumentPlaceholder.Syntax,
+                                            type: argumentPlaceholder.Type,
+                                            constantValue: argumentPlaceholder.GetConstantValue(),
+                                            isImplicit: true));
+                                    }
+                                    else
+                                    {
+                                        Debug.Assert(visitedArguments[argumentPlaceholder.ArgumentIndex] is IFlowCaptureReferenceOperation or IDiscardOperation);
+                                        AddPlaceholder(argumentPlaceholder, visitedArguments[argumentPlaceholder.ArgumentIndex]);
+                                    }
+                                    placeholderKeys.Add(argumentPlaceholder);
+                                    break;
+                            }
+                        }
+                    }
+
+                    IOperation visitedArgument = VisitRequired(originalArgument);
+                    Debug.Assert(visitedArgument is IFlowCaptureReferenceOperation);
+                    visitedArguments.Add(visitedArgument);
```

Comment:
>visitedArguments.Add(visitedArgument); [](http://example.com/codeflow?start=20&length=38)

I think the logic would be simpler to follow if we were to limit the purpose of this method to visiting IInterpolatedStringHandlerCreationOperation argument and returning it. The caller can handle adition of the result to the list.